### PR TITLE
[BE]#Task 210859 Get program data from the program id api changes

### PR DESCRIPTION
**Pull request by GitVivekHub**
https://tracker.tekdi.net/issues/210859

# I have ensured that following `Pull Request Checklist` is taken care of before sending this PR

* [x] Code is formatted as per format decided
* [x] Updated acceptance criteria in tracker
* [x] Updated test cases in test-cases-tracker
* [x] Updated new API endpoint if any in common postman collection
* [x] Updated DB changes in db-tracker if any

## Summary by CodeRabbit

- **Refactor**
	- Enhanced the `getProgramDetails` method to retrieve program and cohort data more efficiently.

### CODE REVIEW

File: src/src/modules/taxonomy/taxonomy.service.ts [hunk 1]
```diff
@@ -3,36 +3,39 @@ import { HasuraService as HasuraServiceFromServices } from '../../services/hasur
...
+			await this.hasuraServiceFromServices.executeRawSql(sql)
+		)?.result;
 
-		if (program_data != null) {
-			return response.status(200).send({
-				success: true,
-				data: program_data,
+		if (cohort_data != undefined) {
+			return response.json({
+				status: 200,
```
Comment (manojLondhe) [inline src/src/modules/taxonomy/taxonomy.service.ts:27]: use `response.status(200)` instead

File: src/src/modules/taxonomy/taxonomy.service.ts [hunk 2]
```diff
@@ -3,36 +3,39 @@ import { HasuraService as HasuraServiceFromServices } from '../../services/hasur
...
+				data: this.hasuraServiceFromServices.getFormattedData(
+					cohort_data,
+					[5],
+				),
 			});
 		} else {
-			return response.status(404).send({
-				success: false,
-				data: {},
+			return response.json({
```
Comment (manojLondhe) [inline src/src/modules/taxonomy/taxonomy.service.ts:35]: should be `response.status(404)`

File: src/src/modules/taxonomy/taxonomy.service.ts [hunk 3]
```diff
@@ -3,36 +3,39 @@ import { HasuraService as HasuraServiceFromServices } from '../../services/hasur
...
-				success: true,
-				data: program_data,
+		if (cohort_data != undefined) {
+			return response.json({
+				status: 200,
+				message: 'Successfully retrieved data',
+				data: this.hasuraServiceFromServices.getFormattedData(
+					cohort_data,
+					[5],
+				),
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/src/modules/taxonomy/taxonomy.service.ts:32]: what are we formatting here?

File: src/src/modules/taxonomy/taxonomy.service.ts (hunk 2 above)
Comment (GitVivekHub) [reply src/src/modules/taxonomy/taxonomy.service.ts:35]: resolved

File: src/src/modules/taxonomy/taxonomy.service.ts (hunk 1 above)
Comment (GitVivekHub) [reply src/src/modules/taxonomy/taxonomy.service.ts:27]: resolved

**Pull request merged**